src/plugins/vis_type_vega/public/vega_inspector/components/signal_viewer.tsx

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React, { useEffect, useState } from 'react';
+
+import { EuiSpacer } from '@elastic/eui';
+import { VegaAdapter, InspectSignalsSets } from '../vega_adapter';
+import { InspectorDataGrid } from './inspector_data_grid';
+
+interface SignalViewerProps {
+  vegaAdapter: VegaAdapter;
+}
+
+export const SignalViewer = ({ vegaAdapter }: SignalViewerProps) => {
+  const [inspectSignalsSets, setInspectSignalsSets] = useState<InspectSignalsSets>();
+
+  useEffect(() => {
+    const subscription = vegaAdapter.getSignalsSetsSubscription().subscribe((signalSets) => {
+      if (signalSets) {
+        setInspectSignalsSets(signalSets);
```

Comment:
The `signals` inspector is not doing what I expected it to do, mostly because it isn't presenting a "live" view of the signals as they pass through the view. Instead, it only presents a snapshot of the signals as they existed when the inspector was switched to the Vega debug view.

This makes the signal viewer impossible to use for the main purpose: viewing the data that's passing through the view during mouse or keyboard interactions.

I don't think the signal viewer is useful until it presents live data using `addSignalListener`: https://vega.github.io/vega/docs/api/view/#view_addSignalListener

Reply:
done 
![Screen Recording 2020-07-16 at 4 39 28 PM](https://user-images.githubusercontent.com/20072247/87679094-6f970480-c784-11ea-9767-be28f8bc62a3.gif)